subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,262 @@
+</sample>
+++++
+
+You may also want to define your own aggregate tasks to help with your workflow. For example, imagine that you have several publications that should be published to the external repository. It could be very useful to publish all of them in one go without publishing the internal ones.
+
+The following sample demonstrates how you can do this by defining an aggregate task — `publishToExternalRepository` — that depends on all the relevant publish tasks:
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="shorthandTasks" title="Defining your own shorthand tasks for publishing">
+    <sourcefile file="build.gradle" snippet="shorthand-tasks"/>
+</sample>
+++++
+
+This particular sample automatically handles the introduction or removal of the relevant publishing tasks by using api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] with the api:org.gradle.api.publish.maven.tasks.PublishToMavenRepository[] task type. You can do the same with api:org.gradle.api.publish.ivy.tasks.PublishToIvyRepository[] if you're publishing to Ivy-compatible repositories.
+
+[[sec:configuring_publishing_tasks]]
+=== Configuring publishing tasks
+
+The publishing plugins create their non-aggregate tasks after the project has been evaluated, which means you cannot directly reference them from your build script. If you would like to configure any of these tasks, you should use deferred task configuration. This can be done in a number of ways via the project's `tasks` collection.
+
+For example, imagine you want to change where the `generatePomFileFor__PubName__Publication` tasks write their POM files. You can do this by using the api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] method, as demonstrated by this sample:
+
+++++
+<sample dir="userguide/publishing/javaLibrary" id="deferred-configuration-publishing-task" title="Configuring a dynamically named task created by the publishing plugins">
+    <sourcefile file="build.gradle" snippet="configure-generate-task"/>
+</sample>
+++++
+
+The above sample uses a regular expression to extract the name of the publication from the name of the task. This is so that there is no conflict between the file paths of all the POM files that might be generated. If you only have one publication, then you don't have to worry about such conflicts since there will only be one POM file.
+
+=== Terminology
+
+[[glossary:artifact]]
+Artifact::
+A file produced by a build, such as a JAR, a ZIP distribution, or a native executable.
++
+Artifacts are typically designed to be used or consumed by users or other projects, or deployed to hosting systems. This is distinct from intermediate files that are created by a build as a step on the way to producing the end result. For example, the class files generated by compiling Java source are almost never an end product themselves, since they are typically packaged into JARs or some other type of archive.
```

Comment:
Would you class it as an artifact? It's really a bunch of files rather than a directory. I guess it is something that can be added to an outgoing configuration and added to a classpath.

Is the definition given reasonable if one excludes the example? Or does the definition itself need to be changed? I don't want to end up with something that's so generic it's almost meaningless or is simply too hard for users to understand.

Reply:
Well, the whole directory is one artifact. You can't publish it, but it's super useful cross-project. So yes your explanation is correct, it's just the example that was a bit unfortunate :)

Reply:
- [X] Make this technically accurate